sorc/regrid_nemsio.fd/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+list(APPEND fortran_src
+constants.f90
+fv3_interface.f90
+gfs_nems_interface.f90
+interpolation_interface.f90
+kinds.f90
+main.f90
+mpi_interface.f90
+namelist_def.f90
+netcdfio_interface.f90
+physcons.f90
+regrid_nemsio_interface.f90
+variable_interface.f90
+)
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fp-model precise")
+endif()
+
+set(exe_name regrid_nemsio.x)
+add_executable(${exe_name} ${fortran_src})
+target_link_libraries(
+  ${exe_name}
+  nemsio::nemsio
+  bacio::bacio_4
+  sp::sp_d
+  w3nco::w3nco_d
+  NetCDF::NetCDF_Fortran
+  MPI::MPI_Fortran)
```

Comment:
We typically define `FC=mpif90` instead of `FC=ifort`.
For a program `mpif90` will take care of linking the appropriate mpi library when building.
When `ifort` is used, it will require the necessary MPI libraries to be linked at compile time.
There is no "rule" to this.
In the current sorc, there are serial programs and MPI programs.  It is "unfair" to compile the serial programs with `mpif90`.  This explicit linking takes care of it and allows building serial and parallel programs as they should.